### PR TITLE
Add bonsai-limit-trie-logs options and update how to page

### PR DESCRIPTION
**Pull request by siladu**
Fixes https://github.com/hyperledger/besu-docs/issues/1624

### CODE REVIEW

**Comment by joaniefromtheblock** [comment]
Thank you @siladu for opening this 🎉

File: docs/public-networks/how-to/bonsai-limit-trie-logs.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ tags:
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 
-To decrease the database size when using the [Bonsai Trie](../concepts/data-storage-formats#bonsai-tries) data storage format, enable the early access feature `--Xbonsai-limit-trie-logs-enabled`. 
+Since Besu version 24.6.0, when using the [Bonsai Trie](../concepts/data-storage-formats#bonsai-tries) data storage format, [`--bonsai-limit-trie-logs-enabled`](../reference/cli/options.md#bonsai-limit-trie-logs-enabled) is enabled by default, unless [`--sync-mode=FULL`](../reference/cli/options.md#sync-mode) in which case this option is disallowed and must be set to false.
```
Comment (joaniefromtheblock) [inline docs/public-networks/how-to/bonsai-limit-trie-logs.md:12]: I don't think we have to include the version; maybe we just add it to the version if it crucial to mention. It most likely is also in the changeling. 

```suggestion
When using the [Bonsai Trie](../concepts/data-storage-formats#bonsai-tries) data storage format, [`--bonsai-limit-trie-logs-enabled`](../reference/cli/options.md#bonsai-limit-trie-logs-enabled) is enabled by default. 
If [`--sync-mode=FULL`](../reference/cli/options.md#sync-mode) is set,  this option is disallowed and must be set to false.
```

File: docs/public-networks/how-to/bonsai-limit-trie-logs.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ The following commands are examples. Before executing these example commands on
 
 :::
 
-1. Add the `--Xbonsai-limit-trie-logs-enabled` option to the [Besu configuration file](use-configuration-file).
+1. Upgrade to Besu version 24.6.0 or add the `--Xbonsai-limit-trie-logs-enabled` option to the [Besu configuration file](use-configuration-file).
```
Comment (joaniefromtheblock) [inline docs/public-networks/how-to/bonsai-limit-trie-logs.md:23]: I think this is a preq and not part of the step

Comment (joaniefromtheblock) [reply docs/public-networks/how-to/bonsai-limit-trie-logs.md:23]: Maybe just a note above with the following:

```
:::note
You are using Besu version 24.6.0 or later
::::
```

or 

```
Ensure you are using Besu version 24.6.0 or later
```

File: docs/public-networks/how-to/bonsai-limit-trie-logs.md (hunk 1 above)
Comment (alexandratran) [reply docs/public-networks/how-to/bonsai-limit-trie-logs.md:12]: So, as per Joan's recommendation, add the new docs to:

- `/docs/public-networks/how-to/bonsai-limit-trie-logs.md`
- `/docs/public-networks/reference/cli/options.md`
- `/versioned_docs/version-24.6.0/public-networks/how-to/bonsai-limit-trie-logs.md`
- `/versioned_docs/version-24.6.0/public-networks/reference/cli/options.md`

And add a note or prerequisite confirming the user is on the correct version.

@siladu let us know if this makes sense to you and we can make the appropriate changes.

Comment (siladu) [reply docs/public-networks/how-to/bonsai-limit-trie-logs.md]: Yep, that makes sense :)

File: docs/public-networks/how-to/bonsai-limit-trie-logs.md (hunk 2 above)
Comment (siladu) [reply docs/public-networks/how-to/bonsai-limit-trie-logs.md:27]: With this one it's either/or: if you're on 24.6.0 it's enabled by default so you don't need to add the flag. If you're on an older version, you must use the experimental form of the option, --Xbonsai-limit-trie-logs-enabled

Comment (siladu) [reply docs/public-networks/how-to/bonsai-limit-trie-logs.md:27]: Maybe it's fine to assume they are on 24.6.0+ if they're reading this version of the docs, in which case we could remove the step entirely.

Comment (siladu) [reply docs/public-networks/how-to/bonsai-limit-trie-logs.md]: I've removed this step and added the note as a prerequsite [`5a93729` (#1627)](https://github.com/hyperledger/besu-docs/pull/1627/commits/5a9372992fa9ea60fb9592ea9f9e7b1a0ae384d2)

**Comment by siladu** [comment]
Added another intro/prerequisite line here: [`d8299f5` (#1627)](https://github.com/hyperledger/besu-docs/pull/1627/commits/d8299f5faba3e4e5cd6d8c596b51ebc1ce279bf9)

Feel free to change/restructure it however you see fit :)

**Review by alexandratran — APPROVED** [review]

**Pull request merged**